### PR TITLE
add yodeswap

### PR DESCRIPTION
**Pull request by dtmkeng**

### CODE REVIEW

**Comment by llamatester** [comment]
The yodeswap adapter exports pools: 
        
 
 Test Suites: 1 passed, 1 total
    Tests:       74 passed, 74 total
    Snapshots:   0 total
    Time:        0.498 s
    Ran all test suites.
    
    Nb of pools: 10
     
    
    Sample pools:
    ┌─────────┬──────────────────────────────────────────────┬─────────────┬────────────┬────────────────┬────────────────────┬────────────────────┬────────────────────┬────────────────────────────────────────────────────────────────────────────────────────────────┬──────────────────────────────────────────────────┐
    │ (index) │                     pool                     │    chain    │  project   │     symbol     │       tvlUsd       │      apyBase       │     apyReward      │                                        underlyingTokens                                        │                   rewardTokens                   │
    ├─────────┼──────────────────────────────────────────────┼─────────────┼────────────┼────────────────┼────────────────────┼────────────────────┼────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────┼──────────────────────────────────────────────────┤
    │    0    │ '0xD95b66FBD36C5754FE7C837500cccBf113fD6ac6' │ 'Dogechain' │ 'yodeswap' │  'USDC-USDT'   │ 1273558.766169952  │ 1.5735446045916472 │ 53.363903383808285 │ [ '0x765277EebeCA2e31912C9946eAe1021199B39C61', '0xE3F5a90F9cb311505cd691a46596599aA1A0AD7D' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    1    │ '0xe180b62C4CF96112e7999200EF9275F83F7E3088' │ 'Dogechain' │ 'yodeswap' │  'BUSD-USDC'   │ 1126050.3207318608 │ 3.226880459801414  │ 50.29531518280965  │ [ '0x332730a4F6E03D9C55829435f10360E13cfA41Ff', '0x765277EebeCA2e31912C9946eAe1021199B39C61' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    2    │ '0x8DCeBE9f071562D52b5ABB17235f3bCA768C1e44' │ 'Dogechain' │ 'yodeswap' │ 'USDC-WWDOGE'  │ 1046496.2900708253 │ 17.160286075080823 │ 227.29868857359773 │ [ '0x765277EebeCA2e31912C9946eAe1021199B39C61', '0xB7ddC6414bf4F5515b52D8BdD69973Ae205ff101' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    3    │ '0x7a3542fADA818C705E1Ce6dA508F5aa5dfC35edd' │ 'Dogechain' │ 'yodeswap' │   'USDC-ETH'   │ 620660.0737596969  │ 3.9798360951312244 │ 91.24971653137209  │ [ '0x765277EebeCA2e31912C9946eAe1021199B39C61', '0xB44a9B6905aF7c801311e8F4E76932ee959c663C' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    4    │ '0xF4887cEf251E6BAfed9478f115B92aebA81d88F7' │ 'Dogechain' │ 'yodeswap' │   'BUSD-BNB'   │  552962.746200745  │         0          │ 122.90532665797846 │ [ '0x332730a4F6E03D9C55829435f10360E13cfA41Ff', '0xA649325Aa7C5093d12D6F98EB4378deAe68CE23F' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    5    │ '0x5aC5529f6d948d565A7832B0950592845B029584' │ 'Dogechain' │ 'yodeswap' │  'USDC-WBTC'   │ 525159.0208361612  │ 0.1770003114021188 │ 150.98108375598966 │ [ '0x765277EebeCA2e31912C9946eAe1021199B39C61', '0xfA9343C3897324496A05fC75abeD6bAC29f8A40f' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    6    │ '0x2Dd53abcFEBE71a9Cd5dFF4f44458E14707F9280' │ 'Dogechain' │ 'yodeswap' │ 'YODE-WWDOGE'  │ 124649.4359492715  │         0          │ 1999.160634711824  │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01', '0xB7ddC6414bf4F5515b52D8BdD69973Ae205ff101' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    7    │ '0xd2333137d28710acEd9E8572D8bf4Baf184e6d16' │ 'Dogechain' │ 'yodeswap' │  'DC-WWDOGE'   │ 106236.14054704769 │ 60.84012013287631  │ 1066.2107170173688 │ [ '0x7B4328c127B85369D9f82ca0503B000D09CF9180', '0xB7ddC6414bf4F5515b52D8BdD69973Ae205ff101' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    8    │ '0x0a5a84AdEE386d295326D93403A92a128130F944' │ 'Dogechain' │ 'yodeswap' │  'YODE-USDC'   │ 41529.07217670491  │ 15.414954928801613 │ 1909.242705271753  │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01', '0x765277EebeCA2e31912C9946eAe1021199B39C61' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    9    │ '0xAE89D749a1EDE455510c86E781B5F6E99A186F1F' │ 'Dogechain' │ 'yodeswap' │ 'DAOGE-WWDOGE' │ 5448.0267545888155 │ 21.991226737317078 │ 2079.103438514127  │ [ '0x7FD7fc6f3dE86b0aC42c76F555C8A55587630494', '0xB7ddC6414bf4F5515b52D8BdD69973Ae205ff101' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    └─────────┴──────────────────────────────────────────────┴─────────────┴────────────┴────────────────┴────────────────────┴────────────────────┴────────────────────┴────────────────────────────────────────────────────────────────────────────────────────────────┴──────────────────────────────────────────────────┘

**Comment by slasher125** [comment]
gm, quick q: seems like the reward apy is ~2x compared to whats being reported on yodeswaps frontend. could u clafiry that pls? also, i'd suggest to move to 24h periods for the fee based apy to be consistent with other projects
ty ser

**Comment by llamatester** [comment]
The yodeswap adapter exports pools: 
        
 
 Test Suites: 1 passed, 1 total
    Tests:       74 passed, 74 total
    Snapshots:   0 total
    Time:        0.417 s
    Ran all test suites.
    
    Nb of pools: 10
     
    
    Sample pools:
    ┌─────────┬──────────────────────────────────────────────┬─────────────┬────────────┬────────────────┬────────────────────┬────────────────────┬────────────────────┬────────────────────────────────────────────────────────────────────────────────────────────────┬──────────────────────────────────────────────────┐
    │ (index) │                     pool                     │    chain    │  project   │     symbol     │       tvlUsd       │      apyBase       │     apyReward      │                                        underlyingTokens                                        │                   rewardTokens                   │
    ├─────────┼──────────────────────────────────────────────┼─────────────┼────────────┼────────────────┼────────────────────┼────────────────────┼────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────┼──────────────────────────────────────────────────┤
    │    0    │ '0xD95b66FBD36C5754FE7C837500cccBf113fD6ac6' │ 'Dogechain' │ 'yodeswap' │  'USDC-USDT'   │ 1392986.722993187  │ 1.340192741733413  │ 32.307708520487225 │ [ '0x765277EebeCA2e31912C9946eAe1021199B39C61', '0xE3F5a90F9cb311505cd691a46596599aA1A0AD7D' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    1    │ '0xe180b62C4CF96112e7999200EF9275F83F7E3088' │ 'Dogechain' │ 'yodeswap' │  'BUSD-USDC'   │ 1183807.2189731204 │ 2.950798890465447  │ 31.680417989574163 │ [ '0x332730a4F6E03D9C55829435f10360E13cfA41Ff', '0x765277EebeCA2e31912C9946eAe1021199B39C61' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    2    │ '0x8DCeBE9f071562D52b5ABB17235f3bCA768C1e44' │ 'Dogechain' │ 'yodeswap' │ 'USDC-WWDOGE'  │ 1062777.1186632644 │ 15.937198855205775 │ 148.21050322001872 │ [ '0x765277EebeCA2e31912C9946eAe1021199B39C61', '0xB7ddC6414bf4F5515b52D8BdD69973Ae205ff101' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    3    │ '0xF4887cEf251E6BAfed9478f115B92aebA81d88F7' │ 'Dogechain' │ 'yodeswap' │   'BUSD-BNB'   │ 571039.1727262271  │         0          │ 78.81107140954217  │ [ '0x332730a4F6E03D9C55829435f10360E13cfA41Ff', '0xA649325Aa7C5093d12D6F98EB4378deAe68CE23F' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    4    │ '0x7a3542fADA818C705E1Ce6dA508F5aa5dfC35edd' │ 'Dogechain' │ 'yodeswap' │   'USDC-ETH'   │ 536864.2825492091  │ 4.296175010408184  │  69.8565889652125  │ [ '0x765277EebeCA2e31912C9946eAe1021199B39C61', '0xB44a9B6905aF7c801311e8F4E76932ee959c663C' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    5    │ '0x5aC5529f6d948d565A7832B0950592845B029584' │ 'Dogechain' │ 'yodeswap' │  'USDC-WBTC'   │ 525853.7179231263  │ 0.1794494148764815 │ 99.84698925391439  │ [ '0x765277EebeCA2e31912C9946eAe1021199B39C61', '0xfA9343C3897324496A05fC75abeD6bAC29f8A40f' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    6    │ '0x2Dd53abcFEBE71a9Cd5dFF4f44458E14707F9280' │ 'Dogechain' │ 'yodeswap' │ 'YODE-WWDOGE'  │ 150001.88999522396 │         0          │ 1100.089026053517  │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01', '0xB7ddC6414bf4F5515b52D8BdD69973Ae205ff101' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    7    │ '0xd2333137d28710acEd9E8572D8bf4Baf184e6d16' │ 'Dogechain' │ 'yodeswap' │  'DC-WWDOGE'   │ 101630.9039521285  │ 68.42783501981184  │ 738.0335322769379  │ [ '0x7B4328c127B85369D9f82ca0503B000D09CF9180', '0xB7ddC6414bf4F5515b52D8BdD69973Ae205ff101' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    8    │ '0x0a5a84AdEE386d295326D93403A92a128130F944' │ 'Dogechain' │ 'yodeswap' │  'YODE-USDC'   │ 46453.90327600572  │ 11.68625642896552  │ 1130.2583167369933 │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01', '0x765277EebeCA2e31912C9946eAe1021199B39C61' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    9    │ '0xAE89D749a1EDE455510c86E781B5F6E99A186F1F' │ 'Dogechain' │ 'yodeswap' │ 'DAOGE-WWDOGE' │ 6059.8657188243515 │ 20.98720371956929  │ 1237.7669491798474 │ [ '0x7FD7fc6f3dE86b0aC42c76F555C8A55587630494', '0xB7ddC6414bf4F5515b52D8BdD69973Ae205ff101' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    └─────────┴──────────────────────────────────────────────┴─────────────┴────────────┴────────────────┴────────────────────┴────────────────────┴────────────────────┴────────────────────────────────────────────────────────────────────────────────────────────────┴──────────────────────────────────────────────────┘

**Comment by dtmkeng** [comment]
I try to fix it. I think it's caused by Block Per Rewards Not being calculated in conjunction with block time.

**Comment by llamatester** [comment]
The yodeswap adapter exports pools: 
        
 
 Test Suites: 1 passed, 1 total
    Tests:       74 passed, 74 total
    Snapshots:   0 total
    Time:        0.406 s
    Ran all test suites.
    
    Nb of pools: 10
     
    
    Sample pools:
    ┌─────────┬──────────────────────────────────────────────┬─────────────┬────────────┬────────────────┬────────────────────┬─────────────────────┬────────────────────┬────────────────────────────────────────────────────────────────────────────────────────────────┬──────────────────────────────────────────────────┐
    │ (index) │                     pool                     │    chain    │  project   │     symbol     │       tvlUsd       │       apyBase       │     apyReward      │                                        underlyingTokens                                        │                   rewardTokens                   │
    ├─────────┼──────────────────────────────────────────────┼─────────────┼────────────┼────────────────┼────────────────────┼─────────────────────┼────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────┼──────────────────────────────────────────────────┤
    │    0    │ '0xD95b66FBD36C5754FE7C837500cccBf113fD6ac6' │ 'Dogechain' │ 'yodeswap' │  'USDC-USDT'   │ 1359215.4310863449 │  1.413786783877962  │  22.6682833268701  │ [ '0x765277EebeCA2e31912C9946eAe1021199B39C61', '0xE3F5a90F9cb311505cd691a46596599aA1A0AD7D' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    1    │ '0xe180b62C4CF96112e7999200EF9275F83F7E3088' │ 'Dogechain' │ 'yodeswap' │  'BUSD-USDC'   │ 1155810.1034032635 │  3.131359588091573  │ 22.214635722740006 │ [ '0x332730a4F6E03D9C55829435f10360E13cfA41Ff', '0x765277EebeCA2e31912C9946eAe1021199B39C61' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    2    │ '0x8DCeBE9f071562D52b5ABB17235f3bCA768C1e44' │ 'Dogechain' │ 'yodeswap' │ 'USDC-WWDOGE'  │ 922044.4259586389  │  18.87365828555962  │ 116.95616685421453 │ [ '0x765277EebeCA2e31912C9946eAe1021199B39C61', '0xB7ddC6414bf4F5515b52D8BdD69973Ae205ff101' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    3    │ '0x5aC5529f6d948d565A7832B0950592845B029584' │ 'Dogechain' │ 'yodeswap' │  'USDC-WBTC'   │  732331.490625907  │ 0.15885293675033435 │ 49.08468505942561  │ [ '0x765277EebeCA2e31912C9946eAe1021199B39C61', '0xfA9343C3897324496A05fC75abeD6bAC29f8A40f' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    4    │ '0xF4887cEf251E6BAfed9478f115B92aebA81d88F7' │ 'Dogechain' │ 'yodeswap' │   'BUSD-BNB'   │ 576564.1091379261  │          0          │ 53.43912325758122  │ [ '0x332730a4F6E03D9C55829435f10360E13cfA41Ff', '0xA649325Aa7C5093d12D6F98EB4378deAe68CE23F' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    5    │ '0x7a3542fADA818C705E1Ce6dA508F5aa5dfC35edd' │ 'Dogechain' │ 'yodeswap' │   'USDC-ETH'   │ 547164.8466965003  │ 4.3797346585179415  │ 46.92534720895143  │ [ '0x765277EebeCA2e31912C9946eAe1021199B39C61', '0xB44a9B6905aF7c801311e8F4E76932ee959c663C' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    6    │ '0x2Dd53abcFEBE71a9Cd5dFF4f44458E14707F9280' │ 'Dogechain' │ 'yodeswap' │ 'YODE-WWDOGE'  │ 117354.06113290125 │          0          │ 962.6762015830823  │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01', '0xB7ddC6414bf4F5515b52D8BdD69973Ae205ff101' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    7    │ '0xd2333137d28710acEd9E8572D8bf4Baf184e6d16' │ 'Dogechain' │ 'yodeswap' │  'DC-WWDOGE'   │ 92744.27834044209  │  79.1719749278877   │ 553.6923866616518  │ [ '0x7B4328c127B85369D9f82ca0503B000D09CF9180', '0xB7ddC6414bf4F5515b52D8BdD69973Ae205ff101' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    8    │ '0x0a5a84AdEE386d295326D93403A92a128130F944' │ 'Dogechain' │ 'yodeswap' │  'YODE-USDC'   │ 39243.650232129556 │ 15.355643290080009  │ 915.9764793500892  │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01', '0x765277EebeCA2e31912C9946eAe1021199B39C61' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    │    9    │ '0xAE89D749a1EDE455510c86E781B5F6E99A186F1F' │ 'Dogechain' │ 'yodeswap' │ 'DAOGE-WWDOGE' │ 5951.021543955529  │ 21.734675041644312  │ 862.9073251413464  │ [ '0x7FD7fc6f3dE86b0aC42c76F555C8A55587630494', '0xB7ddC6414bf4F5515b52D8BdD69973Ae205ff101' ] │ [ '0x6FC4563460d5f45932C473334d5c1C5B4aEA0E01' ] │
    └─────────┴──────────────────────────────────────────────┴─────────────┴────────────┴────────────────┴────────────────────┴─────────────────────┴────────────────────┴────────────────────────────────────────────────────────────────────────────────────────────────┴──────────────────────────────────────────────────┘

File: src/adaptors/yodeswap/index.js [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+    };
+  });
+
+  return res;
+};
+
+module.exports = {
+  timetravel: false,
+  apy: getApy,
+  url: 'https://yodeswap.dog',
```
Comment (Moameri55) [inline src/adaptors/yodeswap/index.js:303]: Very good

**Pull request merged**